zookeeper-api/src/main/java/org/apache/helix/zookeeper/zkclient/annotation/PreFetchChangedData.java

```diff
@@ -22,8 +22,25 @@
 import java.lang.annotation.Retention;
 import java.lang.annotation.RetentionPolicy;
 
-
+/**
+ * An annotation used to prefetch changed data for
+ * {@link org.apache.helix.zookeeper.zkclient.IZkDataListener} in ZkClient.
+ * By default, prefetch is enabled: when ZkClient handles a data change event,
+ * ZkClient will read data and pass data object to
+ * {@link org.apache.helix.zookeeper.zkclient.IZkDataListener#handleDataChange(String, Object)}.
+ * If disabled({@code false}): ZkClient will not read data, so data object is passed as null.
+ * <p>
+ * Example:
+ * <pre>
+ * {@code @PreFetch(enabled = false)}
+ *  public class MyCallback implements IZkDataListener
+ * </pre>
+ *
+ * {@code @PreFetch(enabled = false)} means data will not be prefetched in ZkClient before
+ * handling data change: data is null in
+ * {@link org.apache.helix.zookeeper.zkclient.IZkDataListener#handleDataChange(String, Object)}
+ */
 @Retention(RetentionPolicy.RUNTIME)
```

Comment:
What is the use for this @retention ?

Reply:
@kaisun2000  It tells the Java compiler and JVM that the annotation should be available via reflection at runtime/class/source code. More details you could check jdk javadoc.

Reply:
Right, is it a must here? I don't see we have something like this in other places, say participant side?

Reply:
By the way, just to confirm. After this fix, PreFetchChangeData is used to ZKClient? How about prefetch, who is using it? can you add comment here?

Reply:
`Prefetch` is Helix's logic, should not be involved here in zkClient. The javadoc should stay in that interface, not this one.